library/include/rocwmma/internal/dpp.hpp

```diff
@@ -155,30 +122,92 @@ namespace rocwmma
+                using InputVecT               = VecT<DataT, VecSize>;
+
+                // Ensure that we can vectorize to B32
+                static_assert(sizeof(InputVecT) % sizeof(uint32_t) == 0,
+                              "VecT size must be a multiple of B32");
+                static_assert(sizeof(B32VecT) == sizeof(InputVecT),
+                              "Unable to vectorize src0 to B32");
+
+                auto op = [](auto&& idx, auto&& v0, auto&& v1) {
+                    // Pair up the b32 vector elements with the appropriate b32 scalar elements.
+                    constexpr auto i              = decay_t<decltype(idx)>::value;
+                    constexpr auto v1InB32VecSize = sizeof(v1) / sizeof(uint32_t);
+                    auto v1InB32Vec = reinterpret_cast<VecT<uint32_t, v1InB32VecSize> const&>(v1);
+                    return DppOp::template exec<WriteRowMask, WriteBankMask, BoundCtrl>(
+                        v0.data[i], v1InB32Vec.data[i % v1InB32VecSize]);
+                };
+
+                // Give the current threadId to the threadCtrl modifier.
+                // Then static unroll with cached modifier.
+                auto result = vector_generator<uint32_t, B32VecSize>()(
+                    op, reinterpret_cast<B32VecT const&>(src0), src1);
+                return reinterpret_cast<InputVecT&>(result);
             }
 
             // Vector as prev
             template <typename DataT, uint32_t VecSize>
             ROCWMMA_DEVICE static inline auto exec(VecT<DataT, VecSize> const& src0,
                                                    VecT<DataT, VecSize> const& src1)
             {
-                return forEach(src0, src1, detail::Seq<VecSize>{});
+                // Reinterpret vector as B32 so we can support B64+ elements.
+                constexpr uint32_t B32VecSize = sizeof(DataT) / sizeof(uint32_t) * VecSize;
+                using B32VecT                 = VecT<uint32_t, B32VecSize>;
+                using InputVecT               = VecT<DataT, VecSize>;
+
+                // Ensure that we can vectorize to B32
+                static_assert(sizeof(InputVecT) % sizeof(uint32_t) == 0,
+                              "VecT size must be a multiple of B32");
+                static_assert(sizeof(B32VecT) == sizeof(InputVecT),
+                              "Unable to vectorize src0 to B32");
+
+                auto op = [](auto&& idx, auto&& v0, auto&& v1) {
+                    // Pair up the b32 vector elements with the appropriate b32 scalar elements.
+                    constexpr auto i = decay_t<decltype(idx)>::value;
+                    return DppOp::template exec<WriteRowMask, WriteBankMask, BoundCtrl>(v0.data[i],
+                                                                                        v1.data[i]);
+                };
+
+                // Give the current threadId to the threadCtrl modifier.
```

Comment:
Same with this - there is no threadCtrl cache here

Reply:
deleted.